src/main/java/gift/ProductDao.java

```diff
@@ -0,0 +1,71 @@
+package gift;
+
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.simple.JdbcClient;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class ProductDao {
+    private final JdbcClient jdbcClient;
+
+    public ProductDao(JdbcClient jdbcClient) {
+        this.jdbcClient = jdbcClient;
+    }
+
+    public void save(ProductDto productDto) {
+        jdbcClient.sql("INSERT INTO PRODUCTS (name, price, imageUrl) VALUES (:name, :price, :imageUrl)")
+                .param("name", productDto.getName())
+                .param("price", productDto.getPrice())
+                .param("imageUrl", productDto.getImageUrl())
+                .update();
+    }
+
+
+    public List<Product> getAll() {
+        return jdbcClient.sql("SELECT * FROM PRODUCTS")
+                .query(Product.class)
+                .list();
+    }
+
+    public Product findById(Long id) {
+        try {
+            return jdbcClient.sql("SELECT * FROM PRODUCTS WHERE id = :id")
+                    .param("id", id)
+                    .query(Product.class)
+                    .single();
+        }
+        catch (EmptyResultDataAccessException e) {
+            throw new EmptyResultDataAccessException(1);
+        }
+    }
```

Comment:
```suggestion
    public Product findById(Long id) throws EmptyResultDataAccessException {
        return jdbcClient.sql("SELECT * FROM PRODUCTS WHERE id = :id")
                .param("id", id)
                .query(Product.class)
                .single();
    }
```
EmptyResultDataAccessException을 받아서 그대로 던지는 것 같아서 throws를 써도 되지 않을까? 하는 의견이 있습니다!

Reply:
이 부분은 제가 미처 생각 못했던 부분 같습니다! 수정에 반영하도록 하겠습니다.